src/lib/electricity-maps/index.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable prettier/prettier */
+import * as dotenv from 'dotenv';
+import * as dayjs from 'dayjs';
+import axios from 'axios';
+
+import {ERRORS} from '../../util/errors';
+import {buildErrorMessage} from '../../util/helpers';
+
+import {KeyValuePair, ModelParams} from '../../types/common';
+import {ModelPluginInterface} from '../../interfaces';
+
+
+const {AuthorizationError, InputValidationError, APIRequestError} = ERRORS;
+
+const BASE_URL = 'https://api.electricitymap.org/v3';
+
+export class ElectricityMapsModel implements ModelPluginInterface {
+    authorizationHeader = '';
+    errorBuilder = buildErrorMessage(ElectricityMapsModel);
+    async configure(): Promise<ModelPluginInterface> {
+            this.initAuth();
+            return this;
+    }
```

Comment:
Should configure not have a token parameter that is passed to init auth so there isn't a reliance on ENV variables only?

Reply:
Indeed I considered this but other models do the same, also I'm not sure the interface was meant for this as you would have to pass the token to all your ModelParams which I'm not a big fan of. So I think for now this is good enough ?